com.unity.netcode.gameobjects/Runtime/Messaging/Messages/ParentSyncMessage.cs

```diff
@@ -1,29 +1,43 @@
+using UnityEngine;
+
 namespace Unity.Netcode
 {
     internal struct ParentSyncMessage : INetworkMessage
     {
         public ulong NetworkObjectId;
 
-        public bool IsReparented;
+        public bool WorldPositionStays;
 
         //If(Metadata.IsReparented)
         public bool IsLatestParentSet;
 
         //If(IsLatestParentSet)
         public ulong? LatestParent;
 
+        // These additional properties are used to synchronize clients with the current position
+        // , rotation, and scale after parenting/de-parenting (world/local space relative). This
+        // allows users to control the final child's transform values without having to have
+        // a NetworkTransform component on the child. (i.e. picking something up)
+        // NOTE: Packing and unpacking all serialized properties helps to offset the
+        // additional increased message size.
+        public Vector3 Position;
+        public Quaternion Rotation;
+        public Vector3 Scale;
+
         public void Serialize(FastBufferWriter writer)
         {
-            writer.WriteValueSafe(NetworkObjectId);
-            writer.WriteValueSafe(IsReparented);
-            if (IsReparented)
+            BytePacker.WriteValuePacked(writer, NetworkObjectId);
+            BytePacker.WriteValuePacked(writer, WorldPositionStays);
+            BytePacker.WriteValuePacked(writer, IsLatestParentSet);
+
+            if (IsLatestParentSet)
             {
-                writer.WriteValueSafe(IsLatestParentSet);
-                if (IsLatestParentSet)
-                {
-                    writer.WriteValueSafe((ulong)LatestParent);
-                }
+                BytePacker.WriteValuePacked(writer, (ulong)LatestParent);
             }
+
+            BytePacker.WriteValuePacked(writer, Position);
+            BytePacker.WriteValuePacked(writer, Rotation);
+            BytePacker.WriteValuePacked(writer, Scale);
```

Comment:
Are we actually getting any space savings on these? They're a bunch of unbounded floats and bytepacker just unceremoniously casts the memory location to a UInt8 before trying to pack it. I'd expect this to actually bloat the size of the message since now we're probably serializing an entire Uint8 PLUS a byte for the "packed size" for each of these...